GitFyle.Core.Api/Controllers/ContributionTypesController.cs

```diff
@@ -148,5 +162,49 @@ public async ValueTask<ActionResult<ContributionType>> PutContributionTypeAsync(
             }
         }
 
+        [HttpDelete("{contributionTypeId}")]
+        public async ValueTask<ActionResult<ContributionType>> DeleteContributionTypeByIdAsync(
+            Guid contributionTypeId)
+        {
+            try
+            {
+                ContributionType removedContributionType =
+                    await this.contributionTypeService.RemoveContributionTypeByIdAsync(contributionTypeId);
+
+                return Ok(removedContributionType);
+            }
+            catch (ContributionTypeValidationException contributionTypeValidationException)
+                when (contributionTypeValidationException.InnerException is NotFoundContributionTypeException)
+            {
+                return NotFound(contributionTypeValidationException.InnerException);
+            }
+            catch (ContributionTypeValidationException contributionTypeValidationException)
```

Comment:
Add exception test and catch on the delete for the invalid reference exception.  You will get this exception when you delete the ContributionType, but it has items that reference it.